pkg/phlaredb/block_querier.go

```diff
@@ -767,6 +769,37 @@ func MergeProfilesPprof(ctx context.Context, stream *connect.BidiStream[ingestv1
 	return nil
 }
 
+func Series(ctx context.Context, req *ingestv1.SeriesRequest, blockGetter BlockGetter) (*ingestv1.SeriesResponse, error) {
+	queriers, err := blockGetter(ctx, model.TimeFromUnix(req.Start), model.TimeFromUnix(req.End))
+	if err != nil {
+		return nil, err
+	}
+
+	var labelsSet []*typesv1.Labels
+	var lock sync.Mutex
+	group, ctx := errgroup.WithContext(ctx)
```

Comment:
We probably should bound the amount of parallelism  with some constant.

Reply:
I set a parallelism constant to 50 in https://github.com/grafana/pyroscope/pull/2431/commits/28d98bb50fc695d4c89e2285143ae5ed953778c6. 50 is only a SWAG.